lib/rrdcached_proxy/rrdtool_connection.rb

```diff
@@ -52,12 +53,15 @@ def receive_line(data)
     end
 
     def handle_update(request)
-      return if request.arguments.first =~ @blacklist
+      rrd_file_path = request.arguments.first
+      return if rrd_file_path =~ @blacklist
+
       logger.debug 'Fetching rrd info'
-      field_names = RRDFileInfo.field_names(request.arguments.first)
+      field_names = RRDFileInfo.field_names(rrd_file_path)
 
       logger.debug 'Writing to backend'
-      backend.write UpdateData.new(request, field_names).points
+
+      backend.write UpdateData.new(request, field_names, metadata_regexp).points
```

Comment:
Könntest Du das noch in ein `begin`/`rescue` packen? Damits nicht ganz kracht, wenns kracht.